File: forlab/ForLab.Data/DbModels/CMSSchema/ChannelVideo.cs
﻿using ForLab.Data.BaseModeling;
using System.ComponentModel.DataAnnotations.Schema;

namespace ForLab.Data.DbModels.CMSSchema
{
    [Table("ChannelVideos", Schema = "CMS")]
    public class ChannelVideo : BaseEntity
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public string AttachmentUrl { get; set; }
        public float? AttachmentSize { get; set; }
        public string AttachmentName { get; set; }
        public string ExtensionFormat { get; set; }
        public bool IsExternalResource { get; set; } // false if the user broese and upload the resource itself
    }
}

File: forlab/ForLab.Website/ClientApp/src/@core/models/CMS/ChannelVideo.ts
import { BaseEntityDto } from '../common/BaseEntityDto';
import { BaseFilter } from '../common/BaseFilter';

export class ChannelVideoDto extends BaseEntityDto {
  title: string = null;
  description: string = null;
  attachmentUrl: any = null;
  attachmentSize: number = null;
  attachmentName: string = null;
  extensionFormat: string = null;
  isDefault: boolean = false;
  isExternalResource: boolean = false;
}


export class ChannelVideoFilterDto extends BaseFilter {
  title: string = null;
  attachmentUrl: string = null;
  attachmentSize: number = null;
  attachmentName: string = null;
  extensionFormat: string = null;
  isDefault: boolean = null;
  isExternalResource: boolean = null;
}